not available on armv7l/armhf

Hi,

I wanted to run an executable (a website made with `drame`) built on top of `httpcats` on a machine running an `armv7l` kernel but I got the error:

```shell-session
$ opam install . --deps-only
[...]
[ERROR] Package conflict!
  * Missing dependency:
    - deps-of-drame -> httpcats
    unmet availability conditions: 'arch != "x86_32" & arch != "arm32"'
```

I'm making this in issue in order to track why the library is not available on `arm32`.

Two things:

- first, it seems the constraint has been added in https://github.com/robur-coop/httpcats/commit/23450bb8f5b9b45c2a425024a07a97ea07396982. I have been told it may was because of mirage-crypto, tls or x509;
- then, I also found https://github.com/robur-coop/httpcats/issues/9 which shows that the code may also not be working on 32bits anyway.

I tried to build it by hand to see what happen:

```shell-session
$ git clone https://github.com/robur-coop/httpcats.git
$ cd httpcats
$ opam install . --deps-only
[ERROR] Package conflict!
  * Missing dependency:
    - deps-of-httpcats -> tls-miou-unix >= 1.0.1 -> tls (= 1.0.0 | = 1.0.2 | >= 1.0.4)
    unmet availability conditions, e.g. 'arch != "arm32"'
```

Then:

```shell-session
$ git clone https://github.com/mirleft/ocaml-tls.git
$ cd ocaml-tls
$ opam install ./tls.opam --deps-only
...
Processing 74/77: [mirage-crypto-ec: dune build][1]    23987 killed     opam install ./tls.opam --deps-only
```

I'm currently investigating to provide more information.

So, compiling `mirage-crypto-ec` was reproductibly being OOM-killed (after a really long time!). But for some reason, build with `dune install -p mirage-crypto-ec -j1` was successful (and fast).

I'm back at trying to install `tls`.

So, I managed to install all the dependencies by hand. I had to pass `-j1` to `dune` directly otherwise the build was being killed because of OOM.

The website seems to be working just fine. So I guess the constraint has no reason to be there and can be removed? I tried to run the tests and they are also passing.

--- a/httpcats.opam
+++ b/httpcats.opam
@@ -28,4 +28,4 @@ build: [
   ["dune" "build" "-p" name "-j" jobs]
 ]
 synopsis: "A simple HTTP client using h1, h2, and miou"
-available: [ arch != "x86_32" & arch != "arm32" ]
+available: [ arch != "x86_32" ]

